lib/creator/index.js

```diff
@@ -15,18 +16,30 @@ const runTransform = require('./transformations/index');
 */
 
 function creator(options) {
-	const env = yeoman.createEnv(null, null, new WebpackAdapter());
-	const generatorName = options ? replaceGeneratorName(path.basename(options)) : 'webpack-default-generator';
+	let env = yeoman.createEnv('webpack', null, new WebpackAdapter());
+	const generatorName = options ? replaceGeneratorName(path.basename(options[0])) : 'webpack-default-generator';
 	if(options) {
-		const generatorName = replaceGeneratorName(path.basename(options));
-		env.register(require.resolve(options), generatorName);
+		const WebpackGenerator = class extends Generator {
+			initializing() {
+				options.forEach( (path) => {
+					return this.composeWith(require.resolve(path));
+				});
+			}
+		};
+		env.registerStub(WebpackGenerator, generatorName);
 	} else {
 		env.registerStub(defaultGenerator, 'webpack-default-generator');
 	}
 
 	env.run(generatorName)
 	.on('end', () => {
-		return runTransform(env.getArgument('configuration'));
+		if(generatorName !== 'webpack-default-generator') {
+			//HACK / FIXME
+			env = env.options.env;
+			return runTransform(env.configuration);
```

Comment:
Levels of abstraction in the transform relying on props on `configuration` too, not only webpackOptions.